share/ipld/delete_test.go

```diff
@@ -0,0 +1,101 @@
+package ipld
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/ipfs/boxo/blockstore"
+	"github.com/ipfs/boxo/exchange/offline"
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/sync"
+	"github.com/stretchr/testify/require"
+
+	"github.com/celestiaorg/celestia-node/share/sharetest"
+)
+
+func TestDeleteNode_FullSquare(t *testing.T) {
+	const size = 8
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	bServ := NewMemBlockservice()
```

Comment:
We dont do that anywhere else and its unlikely to give us anything besides test complexity. Delete opereation is the same everywhere

Reply:
Amd making the pruner responsible to delete that data if sampling never succeeds is very debatable. Because that data was never available and once its outside of the window, we like never care about data unavalability and just proceed. Feels a bit wrong. 

Also, making the pruner aware of implementation details of availability feels a bit wrong